File: packages/awesome_notifications_localizations/android/src/main/kotlin/me/carda/awesome_notifications_localizations/LocalizationMethodHandler.kt
package me.carda.awesome_notifications_localizations

import me.carda.android_awn_core.AwesomeMethodHandler

/** Answers the core's setLocalization/getLocalization channel methods. */
class LocalizationMethodHandler(
    private val manager: LocalizationManager
) : AwesomeMethodHandler {

    override fun handle(method: String, arguments: Any?, result: (Any?) -> Unit): Boolean {
        return when (method) {
            "setLocalization" -> {
                result(manager.setLocalization(arguments as? String))
                true
            }
            "getLocalization" -> {
                result(manager.getLocalization())
                true
            }
            else -> false
        }
    }
}

File: packages/awesome_notifications_localizations/ios/awesome_notifications_localizations/Sources/awesome_notifications_localizations/LocalizationMethodHandler.swift
import Foundation
import IosAwnCore

/// Answers the core's setLocalization/getLocalization channel methods, delegated
/// by the base plugin's bridge through the AwesomeMethodHandlerRegistry seam.
final class LocalizationMethodHandler: AwesomeMethodHandler {
    func handle(
        method: String,
        arguments: Any?,
        result: @escaping (Any?) -> Void
    ) -> Bool {
        switch method {
        case "setLocalization":
            result(LocalizationManager.shared.setLocalization(arguments as? String))
            return true
        case "getLocalization":
            result(LocalizationManager.shared.getLocalization())
            return true
        default:
            return false
        }
    }
}
